### DISCUSSION TITLE
Help in setting up wiki.js using docker image on PCF(Pivotal Cloud Foundry)

### DISCUSSION THREAD

**Discussion by deepaklearner** [requarks/wiki#3373] [Help / Questions]
**Question**
Help in setting up wiki.js using docker image on PCF(Pivotal Cloud Foundry)
 db: mysql cleardb

I have created a service key for the database service in PCF. But not able to create the container using the docker image and service key configuration.

**Comment by Smankusors** [comment]
Hmm... What do you mean by "not able to create the container"? Is there any error? Or are you confused how to setup Wiki.js?

**Comment by deepaklearner** [comment]
I am getting two errors:
1. Error in public PCF:  [MASTER] error: Database Initialization Error: create table `analytics` (`key` varchar(255) not null, `isEnabled` boolean not null default '0', `config` json not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json not null) default character set utf8mb4' at line 1

Which I am guessing due to mysql version issue, that I am not sure.

2. Second issue I am getting while setting up in PCF secured inpremise SSL connection - How to setup the docker in SSL secured PCF environment.

**Comment by Smankusors** [comment]
> 1. Error in public PCF:  [MASTER] error: Database Initialization Error: create table `analytics` (`key` varchar(255) not null, `isEnabled` boolean not null default '0', `config` json not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json not null) default character set utf8mb4' at line 1

hmm what version MySQL do you use? It's recommended to use MySQL 8.x for Wiki.js

**Comment by deepaklearner** [comment]
In PCF, they are providing not mysql. Its actually Cleardb, which I dont see in the website of wiki.js in supported list. I think this may be reason. Please can you double check.

**Comment by Smankusors** [comment]
> Cleardb

Cleardb is not the database engine. With Cleardb you can choose from MySQL, PostgreSQL, Oracle, or MSSQL.

**Comment by deepaklearner** [comment]
Yes I chose MySQL in Cleardb. But I got error: "Error in public PCF: [MASTER] error: Database Initialization Error: create table analytics (key varchar(255) not null, isEnabled boolean not null default '0', config json not null) default character set utf8mb4 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json not null) default character set utf8mb4' at line 1"   What should I check now?

**Comment by NGPixel** [comment]
The version of MySQL is most likely too old.

Do yourself a favor and stop using ClearDB. They are extremely overpriced and using outdated tech.
